### PR TITLE
Adds unit tests to useAppState

### PR DESCRIPTION
**Pull request by oidoug**
### Description

Adds unit tests to assert different values expected from `useAppState` hook for different app states.

It uses a `spyOn` mock call to the hook's app state listener, so for each test we need to remove and reattach the spyOn or  the hook is not correctly unmounted between tests.

### CODE REVIEW

File: cardstack/src/hooks/useBiometry.ts [hunk 1]
```diff
@@ -59,10 +59,19 @@ export const useBiometry = () => {
...
         biometryType === SecurityType.BIOMETRIC);
 
+    // Transactions require longpress when biometry is not available or
+    // when using Face ID to minimize user errors.
+    const longPressToConfirm =
+      biometryType === SecurityType.BIOMETRIC ||
+      biometryType === SecurityType.FACE ||
+      biometryType === SecurityType.NONE ||
+      !biometryAvailable;
+
```
Comment (danibonilha) [inline cardstack/src/hooks/useBiometry.ts]: this is probably a conflict, we need to drop

File: cardstack/src/hooks/__tests__/useAppState.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    appStateSpy.mockClear();
+  });
+
+  it('should match values for second open if state changes from background to active', async () => {
+    const { result } = renderHook(() => useAppState());
+    const appStateSpy = jest.spyOn(AppState, 'addEventListener');
+
+    await act(async () => {
+      await appStateSpy.mock.calls[0][1]('background');
+      await appStateSpy.mock.calls[0][1]('active');
```
Comment (danibonilha) [inline cardstack/src/hooks/__tests__/useAppState.test.ts]: supernit: maybe the mockAppState, could be a function that receives the state, but then you'd probably need:
```
  afterEach(() => {
    jest.clearAllMocks();
  });

```

to use a single spy on the describe block

**Review by danibonilha — APPROVED** [review]

File: cardstack/src/hooks/__tests__/useAppState.test.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { waitFor, act } from '@testing-library/react-native';
+import { AppState } from 'react-native';
+
+import { useAppState } from '../../../../src/hooks/useAppState';
```
Comment (oidoug) [inline cardstack/src/hooks/__tests__/useAppState.test.ts]: @danibonilha, this is a bit ugly rn. What do you think is better, to leave the test and hook apart or to move useAppState to cardstack's hooks?

File: cardstack/src/hooks/useBiometry.ts (hunk 1 above)
Comment (oidoug) [reply cardstack/src/hooks/useBiometry.ts]: got it, thanks.

**Pull request merged**